Shared/Samples/Identify KML features/IdentifyKMLFeaturesView.swift

```diff
@@ -0,0 +1,153 @@
+                            try await dataset.load()
+                            
+                            // Create a KML layer with the dataset.
+                            forecastLayer = KMLLayer(dataset: dataset)
+                            
+                            // Add the layer to the map.
+                            map.addOperationalLayer(forecastLayer!)
+                        } catch {
+                            self.error = error
+                        }
+                    }
+            }
+        }
+        .alert(isPresented: $isShowingErrorAlert, presentingError: error)
+    }
+}
+
+private extension IdentifyKMLFeaturesView {
+    /// Identifies the placemark for a given point on the KML layer.
+    /// - Parameters:
+    ///   - screenPoint: The screen point corresponding to a placemark.
+    ///   - proxy: The map view proxy used identify the screen point.
+    /// - Returns: The first KML placemark in the identify result.
+    func kmlPlacemark(for screenPoint: CGPoint, using proxy: MapViewProxy) async throws -> KMLPlacemark? {
+        guard let forecastLayer else {
+            throw CustomError.message("KML layer is not initialized.")
+        }
+        
+        // Identify the screen point on the KML layer using the map view proxy.
+        let identifyResult = try await proxy.identify(on: forecastLayer, screenPoint: screenPoint, tolerance: 2)
+        
+        // Get the first KML placemark from the result's geo elements.
+        let placemark = identifyResult.geoElements.first(where: { $0 is KMLPlacemark })
+        return placemark as? KMLPlacemark
+    }
+    
+    /// Updates the callout text using the balloon content of a given placemark.
+    /// - Parameter placemark: The KML placemark to get the data from.
+    func updateCalloutText(using placemark: KMLPlacemark) throws {
+        // Google Earth only displays the placemarks with description or extended data.
+        // To match its behavior, add a description placeholder if it is empty.
+        if placemark.description.isEmpty {
+            placemark.description = "Weather condition"
+        }
+        
+        // Convert the placemark's html balloon content to text.
+        let data = Data(placemark.balloonContent.utf8)
+        let text = try NSAttributedString(
+            data: data,
+            options: [.documentType: NSAttributedString.DocumentType.html],
+            documentAttributes: nil
+        )
+        
+        // Update the callout text.
+        calloutText = AttributedString(text)
+    }
+    
+    /// An enumeration used to throw an error customized with a string.
+    enum CustomError: LocalizedError {
+        case message(String)
+        
+        /// The text description of the error.
+        var errorDescription: String? {
+            if case .message(let string) = self {
+                return NSLocalizedString(string, comment: "The description of the error thrown.")
```

Comment:
I'm not thinking this is actually localized because the comment for each error message should be different.

Only one case is using this error type and in that location in the code you might consider a precondition instead.